compiler/src/iree/compiler/InputConversion/Common/AutoInputConversionPipeline.cpp

```diff
@@ -99,12 +104,50 @@ static void populateFeatures(Operation *op, const Dialect *chloDialect,
 
 void AutoInputConversionPipelinePass::runOnOperation() {
   ModuleOp module = getOperation();
-  MLIRContext *ctxt = &getContext();
+  MLIRContext *context = &getContext();
+
+  // Check if any plugin-provided pipeline extensions can convert dialects in
+  // the module first.
+  if (pipelineExtensions) {
+    llvm::StringSet<> detectedTypeMnemonics;
+    pipelineExtensions->populateDetectedCustomInputConversionTypes(
+        module, detectedTypeMnemonics);
+
+    if (detectedTypeMnemonics.getNumItems() >= 2) {
+      // TODO(scotttodd): handle multiple typeMnemonics (use all?)
+      module.emitError("not yet implemented mixture of input types");
+      return signalPassFailure();
+    }
+
+    if (!detectedTypeMnemonics.empty()) {
+      auto typeMnemonic = detectedTypeMnemonics.begin()->getKey();
+      OpPassManager passManager(module.getOperationName());
+      bool foundExtension =
+          pipelineExtensions->extendCustomInputConversionPassPipeline(
+              passManager, typeMnemonic);
+      if (!foundExtension) {
+        // We expect that callers properly validate supported extensions and
+        // that if a plugin advertises support, it actually provides it.
+        llvm::errs() << "internal error: custom input conversion for extension "
```

Comment:
Use the usual error reporting mechanism. Spew to llvm::errs(), especially in API cases, often goes to nether regions that are never checked or correlated back to the actual cause of failure.